Find a functionally equivalent code:android.test.ProviderTestCase2.setUp:COMMENT
Method Modifier: protected   
Comment:/**
 * Sets up the environment for the test fixture.
 * <p>
 * Creates a new
 * {@link android.test.mock.MockContentResolver}, a new IsolatedContext
 * that isolates the provider's file operations, and a new instance of
 * the provider under test within the isolated environment.
 * </p>
 *
 * @throws Exception
 */

@@ -6,13 +6,11 @@
     new MockContext2(), // The context that file methods are delegated to
     getContext(), filenamePrefix);
     mProviderContext = new IsolatedContext(mResolver, targetContextWrapper);
-    mProvider = mProviderClass.newInstance();
-    mProvider.attachInfoForTesting(mProviderContext, null);
-    assertNotNull(mProvider);
+    mProvider = createProviderForTest(mProviderContext, mProviderClass, mProviderAuthority);
     mResolver.addProvider(mProviderAuthority, getProvider());
 }

